### PR TITLE
feat(#147): introduces per-pr docs build and tagged versions

### LINKED ISSUE #147: We should version documentation
##### Issue Overview

We don't version documentation. 

##### Expected Behaviour

Each release creates versioned documentation, so it's available under e.g. http://arquillian.org/0.0.1/ (this one I've just manually fixed). We should extend our build script (`.travis.yml`) to perform this for each and every release. Once we have this in place we should retrofit it to other repositories (e.g. [Cube](https://github.com/arquillian/arquillian-cube/tree/gh-pages) was way outdated too, but I fixed that yesterday).

##### Current Behaviour

Only latest documentation is built.

##### Suboptimal fix

For each and every push we can run following script (adjusted to travis environment / docker), but it has obvious disadvantages.

```bash
for tag in `git tag -l`;
do
 git co $tag
 asciidoctor README.adoc -a asciidoctor-source=./docs -o gh-pages/$(git tag --points-at HEAD)/index.html
done
```

And then commit.

### PR DESCRIPTION
**Pull request by bartoszmajsak**
### Features

- builds PR version of the documentation under `arquillian.org/smart-testing/pr/${PR}` and shares the link in the PR comment
- builds tagged doc when tag present on the commit
- cleans up closed PR documentation when on master **and** building the doc (so not every time, but frequently enough)

Documentation generation process is triggered if
- in the range of commits any `.adoc` file was changes
- if the commit is tagged (so we can have for the release commit)
- if commit message contains `[gen docs]` directive

Fixes #147

This crazy amount of commits was for testing travis build... next time I will try with [the local image](https://docs.travis-ci.com/user/common-build-problems/#Troubleshooting-Locally-in-a-Docker-Image).

### CODE REVIEW

**Comment by alien-ike** [comment]
Documentation ready for preview at http://arquillian.org/smart-testing/pr/188

**Comment by alien-ike** [comment]
Documentation ready for preview at http://arquillian.org/smart-testing/pr/188

**Comment by dipak-pawar** [comment]
@bartosz Getting 404

**Comment by alien-ike** [comment]
Documentation ready for preview at http://arquillian.org/smart-testing/pr/188

**Comment by alien-ike** [comment]
Documentation ready for preview at http://arquillian.org/smart-testing/pr/188

**Comment by alien-ike** [comment]
Documentation ready for preview at http://arquillian.org/smart-testing/pr/188

File: .travis.yml [hunk 1]
```diff
@@ -36,30 +38,54 @@ before_script:
...
+      fi
+  '
 
 env:
   - TEST_BED_M2_HOME=/home/travis/.arquillian/mvn/apache-maven-3.3.9
 
 script:
-  - ./mvnw clean verify -P travis -DJAVA_OPTS="-XX:-UseLoopPredicate"
-  - 'if [ $FORCE_DOC_GEN -eq 0 ] || [ $GENERATE_DOC -eq 0 ]; then
+  # - ./mvnw clean verify -P travis -DJAVA_OPTS="-XX:-UseLoopPredicate"
```
Comment (bartoszmajsak) [inline .travis.yml]: Will bring back prior to merge.

**Comment by alien-ike** [comment]
Greeting earthlings! Documentation is ready for preview at http://arquillian.org/smart-testing/pr/188. Just be patient. It might take a few minutes for the changes to show up.

**Comment by alien-ike** [comment]
Greeting earthlings! Documentation is ready for preview at http://arquillian.org/smart-testing/pr/188. Just be patient. It might take a few minutes for the changes to show up.

**Comment by alien-ike** [comment]
Greeting earthlings! Documentation is ready for preview at http://arquillian.org/smart-testing/pr/188. Just be patient. It might take a few minutes for the changes to show up.

**Comment by alien-ike** [comment]
Documentation has been merged with master - latest version can be found at http://arquillian.org/smart-testing/.

**Pull request merged**